lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -381,11 +389,19 @@ export default class DocumentComposer {
 
     didDocument.id = did;
 
-    // Only update the publickey if the array is present
+    // Only update `publickey` if the array is present
```

Comment:
You could include jsonld.js as a dependency, and use `@base` to do this, for all the properties at once... `assertionMethod` (required by the VC Data Model), `authentication` (required by CHAPI), keyAgreementKey, (required by EDVs)...

This solution looks a little brittle to me, but it gets the job done for the current set of features that ION DIDs support.

Reply:
Created issue #512 to rework the transform logic.